framework/include/utils/ADifyFunctor.h

```diff
@@ -0,0 +1,119 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseFunctor.h"
+
+namespace Moose
+{
+template <typename T>
+class ADifyFunctor : public FunctorBase<T>
+{
+public:
+  using typename Moose::FunctorBase<T>::ValueType;
+  using typename Moose::FunctorBase<T>::GradientType;
+  using typename Moose::FunctorBase<T>::DotType;
+
+  ADifyFunctor(const FunctorBase<typename MetaPhysicL::RawType<T>::value_type> & non_ad_functor)
```

Comment:
you used to be against that vehemently?

Reply:
As you get older you get softer? I don't know what to say haha. Maybe I'm amenable because no one has shot themselves in the foot yet...and maybe after a bunch of people shoot themselves in the foot then I'll be back to my old thought patterns? I do think that the general knowledge that the aux system drops derivatives and that the material system keeps derivatives has percolated more through the user base. Maybe that's my justification

Reply:
I guess as things stand now users can already shoot themselves in the foot since aux variables are AD functors

Reply:
Maybe the immediate INL user base has understood that but it s come up in Discussions. People trying to use auxvariables to get going with some variable conversions from vector to components in NS for example

Reply:
In the SubProblem I check to see whether someone has requested a non-AD functor in an AD object when the "true" functor is AD. If so, then I error. If someone requests a non-AD functor in a non-AD object when the "true" functor is AD, then the RawValueFunctor is used